### PR TITLE
add wrapper for zustand persist-middleware

### PR DESCRIPTION
**Pull request by ahmetbicer**
Added wrapper for using MMKV with zustand.

### CODE REVIEW

File: docs/WRAPPER_ZUSTAND_PERSIST_MIDDLEWARE.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      return Promise.resolve(value);
+    } else {
+      return null;
+    }
+  },
+  removeItem: (name) => {
+    storage.delete(name);
+    return Promise.resolve();
+  },
+};
```
Comment (mrousavy) [inline docs/WRAPPER_ZUSTAND_PERSIST_MIDDLEWARE.md:28]: Could you remove all semicolons (`;`)?

File: docs/WRAPPER_ZUSTAND_PERSIST_MIDDLEWARE.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+
+# zustand persist-middleware wrapper
+
+If you want to use MMKV with [zustand persist-middleware](https://github.com/pmndrs/zustand#persist-middleware), create the following `storage` object:
+
+```ts
+import {StateStorage} from 'zustand/middleware';
```
Comment (mrousavy) [inline docs/WRAPPER_ZUSTAND_PERSIST_MIDDLEWARE.md:7]:
```suggestion
import { StateStorage } from 'zustand/middleware';
```

File: docs/WRAPPER_ZUSTAND_PERSIST_MIDDLEWARE.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
```
Comment (mrousavy) [inline docs/WRAPPER_ZUSTAND_PERSIST_MIDDLEWARE.md:8]: Add  the import for MMKV please

**Review by mrousavy — CHANGES_REQUESTED** [review]
Looks great! I just left a few small comments for codestyle, please update

File: docs/WRAPPER_ZUSTAND_PERSIST_MIDDLEWARE.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+If you want to use MMKV with [zustand persist-middleware](https://github.com/pmndrs/zustand#persist-middleware), create the following `storage` object:
+
+```ts
+import { StateStorage } from 'zustand/middleware'
+import { MMKV } from 'react-native-mmkv'
+
+const storage = new MMKV()
+
+const zustandStorage: StateStorage = {
+  setItem: async (name, value) => {
```
Comment (mrousavy) [inline docs/WRAPPER_ZUSTAND_PERSIST_MIDDLEWARE.md:13]:
```suggestion
  setItem: (name, value) => {
```

**Comment by mrousavy** [comment]
perfect thank you!

**Comment by ahmetbicer** [comment]
Thank you so much! 🎉🎉

**Pull request merged**